somaticseq/combine_callers.py

```diff
@@ -0,0 +1,633 @@
+
+    if mutect:
+        
+        import vcfModifier.modify_MuTect as mod_mutect
+            
+        if exclusion:
+            mutect_ex = bed_exclude(mutect, exclusion, outdir + os.sep + 'snv.mutect1.ex.vcf')
+            intermediate_files.add(mutect_ex)
+        else:
+            mutect_ex = mutect
+        
+        if inclusion:
+            mutect_in = bed_include(mutect_ex, inclusion, outdir + os.sep + 'snv.mutect1.in.vcf')
+            intermediate_files.add(mutect_in)
+        else:
+            mutect_in = mutect_ex
+        
+        snv_mutect_out = outdir + os.sep + 'snv.mutect1.vcf'
+        mod_mutect.convert(mutect_in, snv_mutect_out, bam)
+        
+        intermediate_files.add(snv_mutect_out)
+        snv_intermediates.append(snv_mutect_out)
+
+    if mutect2:
+        import vcfModifier.modify_ssMuTect2 as mod_mutect2
+        
+        if exclusion:
+            mutect2_ex = bed_exclude(mutect2, exclusion, outdir + os.sep + 'mutect.ex.vcf')
+            intermediate_files.add(mutect2_ex)
+        else:
+            mutect2_ex = mutect2
+        
+        if inclusion:
+            mutect2_in = bed_include(mutect2_ex, inclusion, outdir + os.sep + 'mutect.in.vcf')
+            intermediate_files.add(mutect2_in)
+        else:
+            mutect2_in = mutect2_ex
+        
+        snv_mutect_out   = outdir + os.sep + 'snv.mutect.vcf'
+        indel_mutect_out = outdir + os.sep + 'indel.mutect.vcf'
+        mod_mutect2.convert(mutect2_in, snv_mutect_out, indel_mutect_out)
+        
+        for file_i in snv_mutect_out, indel_mutect_out:
+            intermediate_files.add( file_i )
+            
+        snv_intermediates.append(snv_mutect_out)
+        indel_intermediates.append(indel_mutect_out)
+        intermediate_vcfs['MuTect2']['snv']   = snv_mutect_out
+        intermediate_vcfs['MuTect2']['indel'] = indel_mutect_out
+        
+    if varscan:
+        import vcfModifier.modify_VarScan2 as mod_varscan2
+
+        if exclusion:
+            varscan_ex = bed_exclude(varscan, exclusion, outdir + os.sep + 'varscan.ex.vcf')
+            intermediate_files.add(varscan_ex)
+        else:
+            varscan_ex = varscan
+        
+        if inclusion:
+            varscan_in = bed_include(varscan_ex, inclusion, outdir + os.sep + 'varscan.in.vcf')
+            intermediate_files.add(varscan_in)
+        else:
+            varscan_in = varscan_ex
+
+        snv_temp          = outdir + os.sep + 'snv.varscan.temp.vcf'
+        indel_temp        = outdir + os.sep + 'indel.varscan.temp.vcf'
+        snv_varscan_out   = outdir + os.sep + 'snv.varscan.vcf'
+        indel_varscan_out = outdir + os.sep + 'indel.varscan.vcf'
+
+        splitVcf.split_into_snv_and_indel(varscan_in, snv_temp, indel_temp)
+        mod_varscan2.convert(snv_temp,   snv_varscan_out)
+        mod_varscan2.convert(indel_temp, indel_varscan_out)
+
+        for file_i in snv_temp, indel_temp, snv_varscan_out, indel_varscan_out:
```

Comment:
`intermediate_files.extend([snv_temp, indel_temp, snv_varscan_out, indel_varscan_out])`?

Reply:
I made intermediate_files a set().

Reply:
Got it. Thanks.